ReactViewControl/ReactViewRender.cs

```diff
@@ -394,26 +402,38 @@ private enum LoadStatus {
         }
 
         private void OnWebViewBeforeResourceLoad(WebView.ResourceHandler resourceHandler) {
-            if (resourceHandler.Url.StartsWith(ResourceUrl.CustomScheme + Uri.SchemeDelimiter)) {
-                var customResourceRequested = CustomResourceRequested;
-                if (customResourceRequested != null) {
-                    resourceHandler.BeginAsyncResponse(() => {
-                        var url = resourceHandler.Url;
-                        var response = customResourceRequested(url);
-
-                        if (response != null) {
-                            string extension = null;
-                            if (Uri.TryCreate(url, UriKind.Absolute, out var uri)) {
-                                var path = uri.AbsolutePath;
-                                extension = Path.GetExtension(path).TrimStart('.');
+            var url = resourceHandler.Url;
+            var scheme = url.Substring(0, Math.Max(0, url.IndexOf(Uri.SchemeDelimiter)));
+
+            switch (scheme.ToLowerInvariant()) {
+                case ResourceUrl.CustomScheme:
+                    var customResourceRequestedHandlers = CustomResourceRequested?.GetInvocationList().ToArray();
+                    if (customResourceRequestedHandlers?.Any() == true) {
```

Comment:
Or (customResourceRequestedHandlers?.Any() ?? false) 😄

Reply:
when using this syntax pattern I prefer the == true ... when using the "?? false" I have to thing twice (I have limitations)